### ISSUE TITLE
no-unnecessary-local-variable false positive

### ISSUE THREAD

**Issue by buu700** [Microsoft/tslint-microsoft-contrib#381]
Code like the following incorrectly triggers a lint error:

```
const balls = {
    foo: 'foo',
    bar: () => balls.foo
};

return balls;
```
